extensions/smallrye-health/deployment/src/test/java/io/quarkus/smallrye/health/test/BlockingNonBlockingTest.java

```diff
@@ -0,0 +1,105 @@
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.Liveness;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.runtime.StartupEvent;
+import io.quarkus.test.InMemoryLogHandler;
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+import io.restassured.parsing.Parser;
+import io.smallrye.common.annotation.Blocking;
+import io.smallrye.health.SmallRyeHealthReporter;
+
+public class BlockingNonBlockingTest {
+
+    private static final java.util.logging.Logger rootLogger = java.util.logging.LogManager.getLogManager()
+            .getLogger("io.vertx.core");
+    private static final InMemoryLogHandler inMemoryLogHandler = new InMemoryLogHandler(
+            record -> record.getLevel().intValue() >= Level.WARNING.intValue());
+
+    @BeforeEach
+    public void setLogHandler() {
+        inMemoryLogHandler.getRecords().clear();
+        rootLogger.addHandler(inMemoryLogHandler);
+    }
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(BlockingHealthCheck.class, StartupBean.class)
+                    .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml"));
+
+    @Test
+    public void testRegisterHealthOnBlockingThreadStep1() {
+        // wait for the initial startup health call to finish
+        try {
+            Thread.sleep(5000);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+
+        try {
+            RestAssured.defaultParser = Parser.JSON;
+            // repeat the call a few times since the block isn't always logged
+            for (int i = 0; i < 3; i++) {
+                RestAssured.when().get("/q/health").then()
+                        .body("status", is("UP"),
+                                "checks.status", contains("UP"),
+                                "checks.name", contains("blocking"));
+            }
+        } finally {
+            RestAssured.reset();
+        }
+
+        if (!inMemoryLogHandler.getRecords().isEmpty()) {
+            LogRecord logRecord = inMemoryLogHandler.getRecords().get(0);
+            assertEquals(Level.WARNING, logRecord.getLevel());
+            assertFalse(logRecord.getMessage().contains("has been blocked for"),
+                    "The blocking health check ran on eventloop thread");
+        }
+    }
+
+    @Liveness
+    static final class BlockingHealthCheck implements HealthCheck {
+        @Override
+        public HealthCheckResponse call() {
+            // block for 3s which is more than allowed default blocking duration of eventloop (2s)
```

Comment:
Instead I would you an illegal operation such as:

```
Uni.createFrom().item(42).onItem().delayIt().by(Duration.ofMillis(10)).await().indefinitely();
```

If you are on the event loop, this operation will be rejected as you are not allowed to use `await()`.